Mathlib/LinearAlgebra/Eigenspace/Basic.lean

```diff
@@ -58,227 +58,549 @@ open FiniteDimensional Set
 variable {K R : Type v} {V M : Type w} [CommRing R] [AddCommGroup M] [Module R M] [Field K]
   [AddCommGroup V] [Module K V]
 
+lemma ker_pow_le_ker_pow_of_le (f : End R M) (k l : ℕ) (hkl : k ≤ l) :
+    LinearMap.ker (f ^ k) ≤ LinearMap.ker (f ^ l) := by
+  rw [← pow_sub_mul_pow _ hkl]
+  exact LinearMap.ker_le_ker_comp _ _
```

Comment:
This already exists as `LinearMap.iterateKer` (more precisely as `f.iterateKer.monotone hkl`).